pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -127,6 +129,11 @@
     private final String clientAddress; // IP address only, no port number included
     private final MessageId startMessageId;
 
+    @Getter
+    @Setter
+    private volatile long consumerEpoch = DEFAULT_READ_EPOCH;
+    public static final long DEFAULT_READ_EPOCH = 0L;
```

Comment:
Can we remove this one, just provide a getter method for `consumerEpoch`? 
It looks like:

1. get epoch from consumer
2. use the epoch to read data
3. dispatch messages along with the epoch